File: src/c/pje_0010.c
#include <stdio.h>
#include <stdbool.h>
#include <math.h>

int main() {
	static const MAX = 2000000;
        long primes[MAX/4];
        primes[0] = 2;
        primes[1] = 3;
        long long n = 3;
        long long sum = 5;
        int t = 2;
        while (primes[t-1] < MAX) {
                n += 2;
                int x;
                bool isPrime = true;
                long max = (int) (ceil(sqrt(n)));
                for (x = 0; x < t && primes[x] <= max; x++) {
                        if (n%primes[x] == 0)
                                isPrime = false;
                }
                if (isPrime) {
                        primes[t] = n;
                        t++;
                        sum += n;
                }
        }
        printf("%lld\n", sum);
        return 0;
}

File: src/python/pje_0010.py
import math
import time

start_time = time.time()

primes = [2, 3]
n = 5
t = 0

while n < 2000000:
	i = 1
	is_prime = 1
	cap = int(math.sqrt(n)) + 1
	while i < len(primes) and is_prime == 1 and primes[i] < cap:
		if n % primes[i] == 0: is_prime = 0
		i = i + 1
	if is_prime == 1: primes.append(n)
	n = n + 2
	
for p in primes:
	t = t + p
	
elapsed_time = time.time() - start_time

print "%d found in %ss" % (t, elapsed_time)
